src/lib/markdown_generators.js

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env node
+
+const yaml = require("js-yaml")
+const markdown = require("markdown-builder")
+const titleCase = require("title-case")
+const TurndownService = require("turndown")
+const tables = require("turndown-plugin-gfm").tables
+const turndownService = new TurndownService()
+turndownService.use(tables)
+const { getCourseImageUrl } = require("./helpers")
+
+const generateMarkdownFromJson = courseData => {
+  /*
+    This function takes JSON data parsed from a master.json file and returns markdown data
+    */
+  const markdownData = []
+  let courseHomeMarkdown = generateCourseHomeFrontMatter(courseData)
+  courseHomeMarkdown += generateCourseFeatures(courseData)
+  courseHomeMarkdown += generateCourseCollections(courseData)
+  markdownData.push({
+    name: "_index.md",
+    data: courseHomeMarkdown
+  })
+  let menuIndex = 10
+  courseData["course_pages"].forEach(page => {
+    if (page["text"]) {
+      const pageName = page["short_url"]
+      let courseSectionMarkdown = generateCourseSectionFrontMatter(
+        page["title"],
+        menuIndex
+      )
+      courseSectionMarkdown += generateCourseSectionMarkdown(page, courseData)
+      const sectionData = {
+        name: `sections/${pageName}.md`,
+        data: courseSectionMarkdown
+      }
+      markdownData.push(sectionData)
+      menuIndex += 10
+    }
+  })
+  return markdownData
+}
+
+const generateCourseHomeFrontMatter = courseData => {
+  /*
+      Generate the front matter metadata for the course home page given course_data JSON
+      */
+
+  let courseNumber = courseData["sort_as"]
+  if (courseData["extra_course_number"]) {
+    if (courseData["extra_course_number"]["sort_as_col"]) {
+      courseNumber += ` / ${courseData["extra_course_number"]["sort_as_col"]}`
+    }
+  }
+  const makeTopic = feature => {
```

Comment:
We should move this to the top level of the file, otherwise this will be recreated for each function call for `generateCourseHomeFrontMatter`. We should only define functions inside other functions if there is some context we need to access here.